docs/troubleshooting.md

```diff
@@ -1,162 +1,410 @@
-# Troubleshooting and debugging CernVM-FS
+# Troubleshooting for CernVM-FS
 
-We focus on client-side issues, that is any issues that may arise on the
-client itself or when it connects to a Squid proxy or a Stratum-1.
+When you experience problems with getting access to a CernVM-FS repository,
+it can be tricky to figure out what the actual underlying cause is,
+giving the complexity of a typical CernVM-FS setup.
 
-## Error overview
+In this section we provide some guidelines on how to troubleshoot some potential problems
+you may run into with CernVM-FS.
+We focus on client-side issues: problems that may arise on a CernVM-FS client system itself,
+or with the connection to external servers like a [Squid proxy](appendix/terminology.md#proxy)
+or a [Stratum 1 replica server](appendix/terminology.md#stratum1).
 
-Considering the CernVM-FS client, errors could be related to
+While some commands suggested below that can help to determine the actual problem at hand
+require system administrator privileges (indicated with the use of `sudo`),
+several can also be run as an unprivileged user.
 
-- the installation of the client software
-  (see [Installation issues](#installation-issues)),
-- the configuration of the client,
-  (see [Configuration issues](#configuration-issues)),
-- the current status of the client,
-  (see [Mounting problems](#mounting-problems) and [Cache problems](#cache-problems)),
-- the network connection to a Squid proxy and/or Stratum-1
-  (see [Connectivity issues](#connectivity-issues))
+<p align="center">
+<img src="../img/off_and_on.png" alt="Have you tried turning it off and on again?" width="60%"/></br>
+</p>
+
+!!! note
+
+    In this section, we will continue to use the [EESSI CernVM-FS repository `software.eessi.io`](
+    eessi/high-level-design.md#filesystem_layer) as a running example, but the troubleshooting guidelines
+    or by no means specific to EESSI.
+
+    Make sure you adjust the example commands to the CernVM-FS repository you are using, if needed.
+
+
+## Typical problems
+
+### Error messages
+
+The error messages that you may encounter when accessing a CernVM-FS repository
+are often quite cryptic, especially if you are not very familiar with CernVM-FS, or with
+file systems and networking on Linux systems in general.
+
+Here are a couple of examples:
+
+* The CernVM-FS repository may not be known (yet) on your system, which will result
+  in a (clear) error message like this when you try to access it:
+  ```
+  $ ls /cvmfs/software.eessi.io
+  ls: cannot access '/cvmfs/software.eessi.io': No such file or directory
+  ```
+  
+* You may see errors messages that suggest network connectivity problems, like:
+  ```
+  Failed to discover HTTP proxy servers (23 - proxy auto-discovery failed)
+  ```
+  ```
+  Transport endpoint is not connected
```

Comment:
This usually indicates a fuse crash, not a network issue. I'll open a pr with a suggestion and some typo fixes